hail/python/hailtop/utils/utils.py

```diff
@@ -50,3 +50,28 @@ def __init__(self, parallelism):
 
     async def wait(self):
         await self._done.wait()
+
+
+async def request_retry_transient_errors(f, *args, **kwargs):
+    delay = 0.1
+    while True:
+        try:
+            return await f(*args, **kwargs)
+        except aiohttp.ClientResponseError as e:
+            # 408 request timeout, 503 service unavailable, 504 gateway timeout
+            if e.status == 408 or e.status == 503 or e.status == 504:
+                pass
+            else:
+                raise
+        except aiohttp.ClientOSError as e:
+            if e.errno == errno.ETIMEDOUT:
+                pass
+            else:
+                log.exception('request failed ClientOSError errno {e.errno} {os.strerror(e.errno)}')
+                raise
+        except aiohttp.ServerTimeoutError:
+            pass
```

Comment:
Can we collect a list of the errors seen so far and throw a custom exception that includes them in its printed message in some way? Maybe just the message, not the stack traces? Might be useful to know which of the three cases happened in what order.

Reply:
I pushed a comment for the one I have on hand.  Most of these haven't been observed, but are things we should retyr.